### PR TITLE
ci: add pre commit hook for commit message check

### LINKED ISSUE #32: 016-Feature-Setup-Pre-Commit-Hook-For-Commit-Message

### PR DESCRIPTION
**Pull request by MitchelHsu**
### Description
- Add pre commit hook to husky for commit message linting

### Related Issues
- Fixed #32 

### Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Other (please specify):

### Checklist
- [x] Code follows the style guide.
- [x] I have performed a self-review of my code.
- [x] Documentation has been updated (if necessary).
- [x] Tests have been added or updated.

### Additional Notes
- Some commit tags you can choose from
  - feat
  - fix
  - docs
  - ci
  - style
  - refactor
  - test
  - chore
  - perf
  - build
  - revert

### CODE REVIEW

**Review by Shyam-Sundar-Bharathi — APPROVED** [review]

**Comment by rashi-bhansali** [comment]
Could you please add the different types of tags to choose from (e.g.: feat, chore, ci, etc.) to the "Additional Notes" part of this PR description. The code changes look concise and relevant to me.

**Comment by MitchelHsu** [comment]
> Could you please add the different types of tags to choose from (e.g.: feat, chore, ci, etc.) to the "Additional Notes" part of this PR description. The code changes look concise and relevant to me.

Ok, I think this makes sense for other to reference from. thanks @rashi-bhansali!

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      'always',
+      [
+        'feat',
+        'fix',
+        'docs',
+        'style',
+        'refactor',
+        'test',
+        'chore',
+        'perf',
```
[the remark this answers is not in the document]
Comment (MitchelHsu) [reply commitlint.config.js:15]: performance

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        'ci',
+        'build',
+        'revert',
```
[the remark this answers is not in the document]
Comment (MitchelHsu) [reply commitlint.config.js:18]: I think this is the list recommended in their repo

**Review by rashi-bhansali — APPROVED** [review]

**Pull request merged**